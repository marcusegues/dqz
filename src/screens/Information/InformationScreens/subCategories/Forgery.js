// @flow
import React from 'react';
import type { ComponentType } from 'react';
import { translate } from 'react-i18next';
import type { TFunction } from '../../../../types/generalTypes';
import { verticalScale } from '../../../../styles/Scaling';
import { SubCategoryContainer } from '../../subComponents/SubCategoryContainer';
import { AppInfoSubText } from '../../../AppInfo/subComponents/AppInfoSubText';

type ForgeryInnerProps = {
  t: TFunction,
};

const ForgeryInner = ({ t }: ForgeryInnerProps) => (
  <SubCategoryContainer title={t('forgeryMainText')}>
    <AppInfoSubText text="forgery" style={{ marginTop: verticalScale(10) }} />
  </SubCategoryContainer>
);

export const Forgery = (translate(['information'])(ForgeryInner): ComponentType<
  ForgeryInnerProps
>);
