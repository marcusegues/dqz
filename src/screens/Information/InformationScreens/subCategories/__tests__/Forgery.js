import React from 'react';
// $FlowFixMe
import { shallow } from 'enzyme';
import { Forgery } from '../Forgery';

jest.mock('react-i18next', () => ({
  translate: () => Component => props => <Component t={() => ''} {...props} />,
}));

describe('Forgery', () => {
  test('Forgery renders as expected', () => {
    const wrapper = shallow(<Forgery />);
    expect(wrapper).toMatchSnapshot();
  });
  test('Forgery renders as expected with language', () => {
    const wrapper = shallow(<Forgery i18n={{ language: 'de' }} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('Forgery renders as expected with language with dive()', () => {
    const wrapper = shallow(<Forgery i18n={{ language: 'de', t: 'test' }} />);
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
