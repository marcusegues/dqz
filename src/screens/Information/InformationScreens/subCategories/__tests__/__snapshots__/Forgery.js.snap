// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgery Forgery renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "t": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "t": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Forgery Forgery renders as expected with language 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    i18n={
      Object {
        "language": "de",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "i18n": Object {
        "language": "de",
      },
      "t": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "i18n": Object {
          "language": "de",
        },
        "t": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Forgery Forgery renders as expected with language with dive() 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForgeryInner
    i18n={
      Object {
        "language": "de",
        "t": "test",
      }
    }
    t={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <AppInfoSubText
          greyBox={false}
          style={
            Object {
              "marginTop": 19.61764705882353,
            }
          }
          text=""
        />,
        <CardRowText
          greyBox={false}
          style={
            Object {
              "marginTop": 39.23529411764706,
            }
          }
          text=""
        />,
        <AppInfoSubText
          greyBox={false}
          style={Object {}}
          text=""
        />,
        <CardRowText
          greyBox={false}
          style={
            Object {
              "marginTop": 39.23529411764706,
            }
          }
          text=""
        />,
        <AppInfoSubText
          greyBox={false}
          style={Object {}}
          text=""
        />,
        <AppInfoSubText
          greyBox={false}
          style={
            Object {
              "marginTop": 19.61764705882353,
            }
          }
          text=""
        />,
        <AppInfoLink
          onPress={[Function]}
          style={Object {}}
          text="www.stop-piracy.ch"
        />,
      ],
      "title": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {
            "marginTop": 19.61764705882353,
          },
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {
            "marginTop": 39.23529411764706,
          },
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {},
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {
            "marginTop": 39.23529411764706,
          },
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {},
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {
            "marginTop": 19.61764705882353,
          },
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onPress": [Function],
          "style": Object {},
          "text": "www.stop-piracy.ch",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <AppInfoSubText
            greyBox={false}
            style={
              Object {
                "marginTop": 19.61764705882353,
              }
            }
            text=""
          />,
          <CardRowText
            greyBox={false}
            style={
              Object {
                "marginTop": 39.23529411764706,
              }
            }
            text=""
          />,
          <AppInfoSubText
            greyBox={false}
            style={Object {}}
            text=""
          />,
          <CardRowText
            greyBox={false}
            style={
              Object {
                "marginTop": 39.23529411764706,
              }
            }
            text=""
          />,
          <AppInfoSubText
            greyBox={false}
            style={Object {}}
            text=""
          />,
          <AppInfoSubText
            greyBox={false}
            style={
              Object {
                "marginTop": 19.61764705882353,
              }
            }
            text=""
          />,
          <AppInfoLink
            onPress={[Function]}
            style={Object {}}
            text="www.stop-piracy.ch"
          />,
        ],
        "title": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {
              "marginTop": 19.61764705882353,
            },
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {
              "marginTop": 39.23529411764706,
            },
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {},
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {
              "marginTop": 39.23529411764706,
            },
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {},
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {
              "marginTop": 19.61764705882353,
            },
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onPress": [Function],
            "style": Object {},
            "text": "www.stop-piracy.ch",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
